components/omega/OmegaBuild.cmake

```
@@ -606,14 +606,24 @@ macro(update_variables)
     set(CMAKE_INSTALL_PREFIX ${OMEGA_INSTALL_PREFIX})
   endif()
 
+  if(NOT DEFINED OMEGA_MPI_ON_DEVICE)
+    option(OMEGA_MPI_ON_DEVICE "Allow device buffers in MPI communication (default ON)." ON)
+  endif()
+
   if("${OMEGA_ARCH}" STREQUAL "CUDA")
     option(Kokkos_ENABLE_CUDA "" ON)
     option(Kokkos_ENABLE_CUDA_LAMBDA "" ON)
     add_definitions(-DOMEGA_TARGET_DEVICE)
+    if(OMEGA_MPI_ON_DEVICE AND OMEGA_CIME_MACHINE STREQUAL "pm-gpu")
+      option(Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC "" OFF)
+    endif()
```

To eliminate system-specific handling, we may introduce a new CMake option `option(OMEGA_CUDA_MALLOC_ASYNC "Enable CUDA malloc async support (default OFF)." OFF)`
```suggestion
option(Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC "" OFF)
set(Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC ${OMEGA_CUDA_MALLOC_ASYNC} CACHE BOOL "" FORCE)
```
Because the option is disabled by default, it is compatible with various versions of Kokkos and MPI types. If we want to achieve better performance on a system that supports CUDA malloc async, we can enable the new CMake option via the command line.